thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1230,8 +1229,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
                     // If !preserverOrder, right rows added to single array in jg, so pass 0
                     joinGroup->addRightMatchCompletePending(activity.preserveOrder ? indexPartNo : 0, sequence, fetchRow);
 
-                    diskAccepted++;
-                    if (diskAccepted > activity.rowLimit)
+                    if (++activity.statsArr[AS_DiskAccepted] > activity.rowLimit)
```

Comment:
This is less efficient, but I agree it is needed because you actually care about inter thread consistency.